### ISSUE TITLE
Download batch with retries > 0 and lost network connection

### ISSUE THREAD

**Issue by YankoSoulisimo** [781flyingdutchman/background_downloader#77]
Hi, I am downloading two fairly large files in a batch. The tasks are set with 'retries = 1'. Whenever the internet connection is lost during downloads, a retry is triggered, then new tasks are enqueued. After the connection has resumed, the original two tasks continue, but also the new two ones start. If this scenario repeats a few times, I end up with a bunch of tasks doing the same thing, with a different progress - first two are almost ready, next two just starting, and so on. Is this a bug and how am I supposed to continue with the original tasks only? Code with task and batch:

```
final List<DownloadTask> tasks = [];
for () { // some for loop, that creates and adds two tasks
   final task = DownloadTask(
        url: offlineSection.downloadUrl!,
        filename: path.basename(file.path),
        baseDirectory: BaseDirectory.applicationDocuments,
        updates: Updates.statusAndProgress,
        retries: 1,
        taskId: offlineSection.key,
      );
      
   tasks.add(task);
}

FileDownloader().downloadBatch(tasks,
      taskStatusCallback: _taskStatusCallback,
      taskProgressCallback: _taskProgressCallback);
```

Using:
flutter 3.10.5
platform: Android 12
background_downloader: 7.4.1

**Comment by 781flyingdutchman** [comment]
Hi, thanks for reaching out. A few questions:
* Do you get the same result when you download only one large file? 
* Can you confirm that both the taskId and url of the two tasks are unique for each task?
* Does the same happen on iOS?
* can you share logs?

Thanks!

**Comment by YankoSoulisimo** [comment]
Hi @781flyingdutchman ,
Here are the answers to your questions:
- yes, the result seems to be the same with one file only. The file is 586 MB, to be precise. We download it from Google Cloud Storage, to give you more context. In fact, I've tried a couple of other download options - not only in a batch, the result is the same in this situation.
- I'm not sure what exactly you mean here - the two IDs of the original tasks in the batch are unique, the urls also. The next two tasks appear to be with the same IDs. I can confirm I don't start them myself - the download code is not triggered second time. 
- I added the final result = await ... part, plus I'm logging whenever a task is enqueued:

```
logger.info('Download batch starting');
      final result = await FileDownloader().downloadBatch(_tasks!,
          taskStatusCallback: _taskStatusCallback,
          taskProgressCallback: _taskProgressCallback);
      logger.info('Download batch ready: ${result}');
      
```

- on iOS it behaves as expected - the task(s) is resumed on internet connectivity available and progress continues from where it reached before that.
- Some logs - you can notice the underlying task appears to be cancelled when the connection is lost:

```
I/flutter (14430): 2023-06-28 11:32:42.802 INFO: Download batch starting
I/BackgroundDownloader(14430): Enqueuing task with id oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:42.846 FINE: Migrating Localstore database from version 0 to 1
I/TaskWorker(14430): Starting task with taskId oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:42.922 INFO: Download task: oxf_en_fr enqueued
I/flutter (14430): 2023-06-28 11:32:42.924 INFO: Download task: oxf_en_fr running
D/TaskWorker(14430): Download for taskId oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:42.935 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:44.011 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:44.014 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:44.015 FINER: Download percentage: 2
I/flutter (14430): 2023-06-28 11:32:44.781 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:44.785 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:44.786 FINER: Download percentage: 4
I/flutter (14430): 2023-06-28 11:32:45.503 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:45.507 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:45.509 FINER: Download percentage: 6
I/flutter (14430): 2023-06-28 11:32:46.332 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:46.335 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:46.336 FINER: Download percentage: 8
I/flutter (14430): 2023-06-28 11:32:47.072 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:47.077 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:47.079 FINER: Download percentage: 10
I/flutter (14430): 2023-06-28 11:32:47.765 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:47.768 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:47.769 FINER: Download percentage: 12
I/flutter (14430): 2023-06-28 11:32:48.673 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:48.681 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:48.682 FINER: Download percentage: 14
I/flutter (14430): 2023-06-28 11:32:49.448 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:49.467 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:49.468 FINER: Download percentage: 16
I/flutter (14430): 2023-06-28 11:32:50.129 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:50.136 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:50.137 FINER: Download percentage: 18
I/flutter (14430): 2023-06-28 11:32:50.824 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:50.827 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:50.828 FINER: Download percentage: 20
I/flutter (14430): 2023-06-28 11:32:51.499 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:51.506 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:51.507 FINER: Download percentage: 22
I/flutter (14430): 2023-06-28 11:32:52.219 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:52.226 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:52.227 FINER: Download percentage: 24
I/flutter (14430): 2023-06-28 11:32:52.960 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:52.968 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:52.969 FINER: Download percentage: 26
I/flutter (14430): 2023-06-28 11:32:54.137 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:54.140 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:54.141 FINER: Download percentage: 28
I/flutter (14430): 2023-06-28 11:32:55.098 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:55.101 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:55.102 FINER: Download percentage: 30
I/flutter (14430): 2023-06-28 11:32:55.975 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:55.982 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:55.984 FINER: Download percentage: 32
I/flutter (14430): 2023-06-28 11:32:56.936 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:56.942 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:56.946 FINER: Download percentage: 34
I/flutter (14430): 2023-06-28 11:32:57.611 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:57.614 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:57.614 FINER: Download percentage: 36
I/flutter (14430): 2023-06-28 11:32:58.151 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:58.156 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:58.157 FINER: Download percentage: 38
I/flutter (14430): 2023-06-28 11:32:58.657 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:58.660 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:58.661 FINER: Download percentage: 40
I/flutter (14430): 2023-06-28 11:32:59.352 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:59.355 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:59.357 FINER: Download percentage: 42
I/flutter (14430): 2023-06-28 11:32:59.863 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:32:59.871 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:32:59.873 FINER: Download percentage: 44
I/flutter (14430): 2023-06-28 11:33:00.406 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:00.410 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:00.411 FINER: Download percentage: 46
I/flutter (14430): 2023-06-28 11:33:00.917 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:00.923 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:00.925 FINER: Download percentage: 48
I/flutter (14430): 2023-06-28 11:33:01.460 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:01.462 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:01.463 FINER: Download percentage: 50
I/flutter (14430): 2023-06-28 11:33:02.123 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:02.126 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:02.126 FINER: Download percentage: 52
V/NativeCrypto(14430): Read error: ssl=0x728bfd9408: I/O error during system call, Software caused connection abort
I/TaskWorker(14430): Exception for oxf_en_fr: javax.net.ssl.SSLException: Read error: ssl=0x728bfd9408: I/O error during system call, Software caused connection abort
V/NativeCrypto(14430): Read error: ssl=0x728c642908: I/O error during system call, Software caused connection abort
V/NativeCrypto(14430): SSL shutdown failed: ssl=0x728bfd9408: I/O error during system call, Broken pipe
V/NativeCrypto(14430): Write error: ssl=0x728c642908: I/O error during system call, Broken pipe
V/NativeCrypto(14430): SSL shutdown failed: ssl=0x728c642908: I/O error during system call, Success
W/ManagedChannelImpl(14430): [{0}] Failed to resolve name. status={1}
I/flutter (14430): 2023-06-28 11:33:02.922 INFO: Download task: oxf_en_fr waiting to retry
I/flutter (14430): 2023-06-28 11:33:02.925 FINER: TaskId oxf_en_fr failed, waiting 2 seconds before retrying. 0 retries remaining
I/WM-WorkerWrapper(14430): Work [ id=a9bf6513-1e8e-42ca-b1b1-f2852b03fd32, tags={ taskId=oxf_en_fr, com.bbflight.background_downloader.TaskWorker, group=await, BackgroundDownloader } ] was cancelled
I/WM-WorkerWrapper(14430): java.util.concurrent.CancellationException: Task was cancelled.
I/WM-WorkerWrapper(14430): 	at androidx.work.impl.utils.futures.AbstractFuture.cancellationExceptionWithCause(AbstractFuture.java:1184)
I/WM-WorkerWrapper(14430): 	at androidx.work.impl.utils.futures.AbstractFuture.getDoneValue(AbstractFuture.java:514)
I/WM-WorkerWrapper(14430): 	at androidx.work.impl.utils.futures.AbstractFuture.get(AbstractFuture.java:475)
I/WM-WorkerWrapper(14430): 	at androidx.work.impl.WorkerWrapper$2.run(WorkerWrapper.java:311)
I/WM-WorkerWrapper(14430): 	at androidx.work.impl.utils.SerialExecutor$Task.run(SerialExecutor.java:91)
I/WM-WorkerWrapper(14430): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
I/WM-WorkerWrapper(14430): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
I/WM-WorkerWrapper(14430): 	at java.lang.Thread.run(Thread.java:919)
I/flutter (14430): 2023-06-28 11:33:02.929 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:02.932 FINER: On connectivity changed: ConnectivityResult.none
I/flutter (14430): 2023-06-28 11:33:02.935 FINEST: On network state changed: ConnectivityResult.none
I/flutter (14430): 2023-06-28 11:33:02.936 FINEST: On network state changed: ConnectivityResult.none
I/flutter (14430): 2023-06-28 11:33:02.938 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:02.940 FINER: Download percentage: -400
W/ManagedChannelImpl(14430): [{0}] Failed to resolve name. status={1}
I/BackgroundDownloader(14430): Enqueuing task with id oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:04.987 INFO: Download task: oxf_en_fr enqueued
W/ManagedChannelImpl(14430): [{0}] Failed to resolve name. status={1}
W/ManagedChannelImpl(14430): [{0}] Failed to resolve name. status={1}
W/ManagedChannelImpl(14430): [{0}] Failed to resolve name. status={1}
I/flutter (14430): 2023-06-28 11:33:15.559 FINER: On connectivity changed: ConnectivityResult.wifi
I/flutter (14430): 2023-06-28 11:33:15.562 FINEST: On network state changed: ConnectivityResult.wifi
D/FRCPlugin(14430): Sending fetchTimeout: 10
D/FRCPlugin(14430): Sending fetchTimeout: 10
I/TaskWorker(14430): Starting task with taskId oxf_en_fr
I/flutter (14430):         "is_purchased" = ?
I/TaskWorker(14430): Starting task with taskId oxf_en_fr
I/flutter (14430):       WHERE 
I/flutter (14430):         "key" = ?
I/flutter (14430):        with args [1, oxf_ode]
I/flutter (14430): 2023-06-28 11:33:15.652 INFO: Download task: oxf_en_fr running
I/flutter (14430): 2023-06-28 11:33:15.653 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries"; with args []
I/flutter (14430): 2023-06-28 11:33:15.654 INFO: Download task: oxf_en_fr running
I/DpmTcmClient(14430): RegisterTcmMonitor from: $Proxy0
D/TaskWorker(14430): Download for taskId oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:18.706 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:18.707 FINER: Download percentage: 2
I/flutter (14430): 2023-06-28 11:33:19.754 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:21.480 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:21.483 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:21.484 FINER: Download percentage: 4
I/flutter (14430): 2023-06-28 11:33:22.598 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:23.845 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:23.848 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:23.849 FINER: Download percentage: 6
I/flutter (14430): 2023-06-28 11:33:24.226 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:25.749 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:25.751 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:25.752 FINER: Download percentage: 8
I/flutter (14430): 2023-06-28 11:33:26.059 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:27.628 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:27.636 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:27.638 FINER: Download percentage: 10
I/flutter (14430): 2023-06-28 11:33:28.191 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:28.717 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:28.720 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:28.720 FINER: Download percentage: 12
I/flutter (14430): 2023-06-28 11:33:29.891 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:29.895 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:29.896 FINER: Download percentage: 14
I/flutter (14430): 2023-06-28 11:33:30.557 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:30.559 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:30.560 FINER: Download percentage: 12
I/flutter (14430): 2023-06-28 11:33:31.736 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:31.741 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:31.742 FINER: Download percentage: 16
I/flutter (14430): 2023-06-28 11:33:33.056 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:33.061 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:33.063 FINER: Download percentage: 14
I/flutter (14430): 2023-06-28 11:33:35.790 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:35.795 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:35.795 FINER: Download percentage: 16
I/flutter (14430): 2023-06-28 11:33:35.853 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:35.855 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:35.855 FINER: Download percentage: 18
I/flutter (14430): 2023-06-28 11:33:38.506 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:38.738 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:38.741 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:38.742 FINER: Download percentage: 20
I/flutter (14430): 2023-06-28 11:33:39.950 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:40.291 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:40.298 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:40.300 FINER: Download percentage: 22
I/flutter (14430): 2023-06-28 11:33:41.807 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:43.159 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:43.167 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:43.168 FINER: Download percentage: 24
I/flutter (14430): 2023-06-28 11:33:44.214 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:47.100 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:47.104 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:47.105 FINER: Download percentage: 26
I/flutter (14430): 2023-06-28 11:33:47.869 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:48.914 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:48.917 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:48.917 FINER: Download percentage: 28
I/flutter (14430): 2023-06-28 11:33:50.225 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:50.229 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:50.231 FINER: Download percentage: 30
I/flutter (14430): 2023-06-28 11:33:50.468 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:50.471 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:50.471 FINER: Download percentage: 28
I/flutter (14430): 2023-06-28 11:33:51.561 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:51.563 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:51.564 FINER: Download percentage: 32
I/flutter (14430): 2023-06-28 11:33:54.166 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:54.172 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:54.173 FINER: Download percentage: 30
I/flutter (14430): 2023-06-28 11:33:55.505 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:55.508 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:55.509 FINER: Download percentage: 34
I/flutter (14430): 2023-06-28 11:33:58.152 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:58.155 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:58.156 FINER: Download percentage: 36
I/flutter (14430): 2023-06-28 11:33:58.394 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:58.397 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:58.397 FINER: Download percentage: 32
I/flutter (14430): 2023-06-28 11:33:59.563 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:33:59.570 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:33:59.571 FINER: Download percentage: 38
I/flutter (14430): 2023-06-28 11:34:00.092 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:00.095 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:00.096 FINER: Download percentage: 34
I/flutter (14430): 2023-06-28 11:34:00.500 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:00.507 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:00.509 FINER: Download percentage: 40
I/flutter (14430): 2023-06-28 11:34:01.661 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:01.665 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:01.666 FINER: Download percentage: 42
I/flutter (14430): 2023-06-28 11:34:02.301 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:02.303 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:02.304 FINER: Download percentage: 36
I/flutter (14430): 2023-06-28 11:34:02.512 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:02.514 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:02.514 FINER: Download percentage: 44
I/flutter (14430): 2023-06-28 11:34:03.291 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:03.295 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:03.296 FINER: Download percentage: 46
I/flutter (14430): 2023-06-28 11:34:04.285 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:04.291 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:04.292 FINER: Download percentage: 48
I/flutter (14430): 2023-06-28 11:34:05.114 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:05.116 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:05.117 FINER: Download percentage: 38
I/flutter (14430): 2023-06-28 11:34:05.374 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:05.377 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:05.378 FINER: Download percentage: 50
I/flutter (14430): 2023-06-28 11:34:06.693 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:06.695 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:06.696 FINER: Download percentage: 52
I/flutter (14430): 2023-06-28 11:34:07.504 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:07.508 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:07.509 FINER: Download percentage: 54
I/flutter (14430): 2023-06-28 11:34:08.224 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:08.229 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:08.229 FINER: Download percentage: 40
I/flutter (14430): 2023-06-28 11:34:08.348 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:08.350 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:08.350 FINER: Download percentage: 56
I/flutter (14430): 2023-06-28 11:34:09.667 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:09.674 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:09.675 FINER: Download percentage: 58
I/flutter (14430): 2023-06-28 11:34:10.956 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:10.960 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:10.961 FINER: Download percentage: 60
I/flutter (14430): 2023-06-28 11:34:11.003 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:11.005 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:11.005 FINER: Download percentage: 42
I/flutter (14430): 2023-06-28 11:34:11.943 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:11.949 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:11.951 FINER: Download percentage: 62
I/flutter (14430): 2023-06-28 11:34:12.801 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:12.805 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:12.806 FINER: Download percentage: 44
I/flutter (14430): 2023-06-28 11:34:12.954 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:12.958 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:12.959 FINER: Download percentage: 64
I/flutter (14430): 2023-06-28 11:34:13.951 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:13.955 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:13.956 FINER: Download percentage: 66
I/flutter (14430): 2023-06-28 11:34:14.334 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:14.340 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:14.341 FINER: Download percentage: 46
I/flutter (14430): 2023-06-28 11:34:15.724 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:15.727 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:15.728 FINER: Download percentage: 68
I/flutter (14430): 2023-06-28 11:34:15.736 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:15.738 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:15.739 FINER: Download percentage: 48
I/flutter (14430): 2023-06-28 11:34:16.627 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:16.629 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:16.630 FINER: Download percentage: 50
I/flutter (14430): 2023-06-28 11:34:17.370 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:17.373 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:17.374 FINER: Download percentage: 52
I/flutter (14430): 2023-06-28 11:34:17.406 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:17.409 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:17.409 FINER: Download percentage: 70
I/flutter (14430): 2023-06-28 11:34:18.140 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:18.147 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:18.148 FINER: Download percentage: 54
I/flutter (14430): 2023-06-28 11:34:19.083 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:19.088 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:19.088 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:19.089 FINER: Download percentage: 72
I/flutter (14430): 2023-06-28 11:34:19.092 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:19.092 FINER: Download percentage: 56
I/flutter (14430): 2023-06-28 11:34:20.473 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:20.477 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:20.478 FINER: Download percentage: 58
I/flutter (14430): 2023-06-28 11:34:20.841 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:20.844 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:20.844 FINER: Download percentage: 74
I/flutter (14430): 2023-06-28 11:34:21.571 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:21.576 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:21.578 FINER: Download percentage: 60
I/flutter (14430): 2023-06-28 11:34:22.248 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:22.254 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:22.255 FINER: Download percentage: 76
I/flutter (14430): 2023-06-28 11:34:22.794 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:22.798 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:22.799 FINER: Download percentage: 62
I/flutter (14430): 2023-06-28 11:34:23.564 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:23.568 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:23.572 FINER: Download percentage: 78
I/flutter (14430): 2023-06-28 11:34:24.089 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:24.094 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:24.095 FINER: Download percentage: 64
I/flutter (14430): 2023-06-28 11:34:25.430 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:25.437 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:25.437 FINER: Download percentage: 80
I/flutter (14430): 2023-06-28 11:34:26.280 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:26.282 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:26.283 FINER: Download percentage: 66
I/flutter (14430): 2023-06-28 11:34:27.037 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:27.044 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:27.045 FINER: Download percentage: 82
I/flutter (14430): 2023-06-28 11:34:27.706 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:27.709 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:27.710 FINER: Download percentage: 68
I/flutter (14430): 2023-06-28 11:34:28.809 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:28.812 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:28.813 FINER: Download percentage: 84
I/flutter (14430): 2023-06-28 11:34:28.858 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:28.860 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:28.861 FINER: Download percentage: 70
I/flutter (14430): 2023-06-28 11:34:29.759 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:29.763 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:29.763 FINER: Download percentage: 72
I/flutter (14430): 2023-06-28 11:34:30.619 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:30.622 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:30.623 FINER: Download percentage: 86
I/flutter (14430): 2023-06-28 11:34:31.085 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:31.088 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:31.088 FINER: Download percentage: 74
I/flutter (14430): 2023-06-28 11:34:31.837 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:31.840 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:31.840 FINER: Download percentage: 88
I/flutter (14430): 2023-06-28 11:34:32.337 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:32.340 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:32.342 FINER: Download percentage: 76
I/flutter (14430): 2023-06-28 11:34:33.099 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:33.101 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:33.102 FINER: Download percentage: 90
I/flutter (14430): 2023-06-28 11:34:33.509 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:33.519 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:33.520 FINER: Download percentage: 78
I/flutter (14430): 2023-06-28 11:34:34.366 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:34.369 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:34.370 FINER: Download percentage: 92
I/flutter (14430): 2023-06-28 11:34:34.668 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:34.672 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:34.673 FINER: Download percentage: 80
I/flutter (14430): 2023-06-28 11:34:35.795 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:35.797 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:35.798 FINER: Download percentage: 94
I/flutter (14430): 2023-06-28 11:34:36.092 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:36.094 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:36.095 FINER: Download percentage: 82
I/flutter (14430): 2023-06-28 11:34:36.872 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:36.881 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:36.882 FINER: Download percentage: 96
I/flutter (14430): 2023-06-28 11:34:37.562 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:37.565 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:37.566 FINER: Download percentage: 84
I/flutter (14430): 2023-06-28 11:34:37.978 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:37.980 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:37.981 FINER: Download percentage: 98
I/flutter (14430): 2023-06-28 11:34:38.796 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:38.798 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:38.799 FINER: Download percentage: 86
I/TaskWorker(14430): Successfully downloaded taskId oxf_en_fr to /data/user/0/com.mobisystems.msdict.embedded.wireless.oxford.dictionaryofenglish/app_flutter/oxf_en_fr.db
I/flutter (14430): 2023-06-28 11:34:38.968 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:38.976 INFO: Download task: oxf_en_fr was successful
I/flutter (14430): 2023-06-28 11:34:38.980 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:38.982 FINER: Task Id: oxf_en_fr
I/flutter (14430): 2023-06-28 11:34:38.982 FINER: Download percentage: 100
I/flutter (14430): 2023-06-28 11:34:38.989 INFO: Download batch ready: Instance of 'Batch'
I/WM-WorkerWrapper(14430): Worker result SUCCESS for Work [ id=d2429d69-9bf9-412d-a6a4-458698a687eb, tags={ taskId=oxf_en_fr, com.bbflight.background_downloader.TaskWorker, group=await, BackgroundDownloader } ]
I/flutter (14430): 2023-06-28 11:34:39.049 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries" WHERE "key" = ?; with args [oxf_en_fr]
I/flutter (14430): 2023-06-28 11:34:39.051 FINEST: Drift: Sent SELECT * FROM "offline_dictionaries"; with args []
I/TaskWorker(14430): Successfully downloaded taskId oxf_en_fr to /data/user/0/com.mobisystems.msdict.embedded.wireless.oxford.dictionaryofenglish/app_flutter/oxf_en_fr.db
I/WM-WorkerWrapper(14430): Worker result SUCCESS for Work [ id=a9bf6513-1e8e-42ca-b1b1-f2852b03fd32, tags={ taskId=oxf_en_fr, com.bbflight.background_downloader.TaskWorker, group=await, BackgroundDownloader } ]

```

**Comment by 781flyingdutchman** [comment]
Thanks, this is helpful. The issue is that when the network disconnects, the task properly fails but  the Android WorkManager *also* cancels the task because the constraints are no longer met (in this case: network connected constraint). That triggers a retry when the network reconnects, and therefore both the old task and the rescheduled one run in parallel. What I need to happen is that the WorkManager task doesn't cancel but completes, so that I control the retry fully (this is because there are many reasons why a task can fail that does not trigger a WorkManager cancellation, so I cannot rely on that mechanism for retries in general).

I need to think about this a bit - meanwhile, let me know if you see a way to accomplish this.

**Comment by 781flyingdutchman** [comment]
Hi @YankoSoulisimo I've developed a fix for this issue and was wondering if you could test it before publishing. It's on the `dev` branch, so you'd have to change your `pubspec.yaml` to read:
```
background_downloader:
    git:
      url: https://github.com/781flyingdutchman/background_downloader.git
      ref: dev
```

Let me know if it fixes the problem of the 'double download'. Note that while it fixes that issue all the time, it's not perfect in other ways: I see instances where the progress value of -1.0 is sent (indicating failure of a download when the network goes down) but not the `.failed` status update (because the WorkManager task is killed by Android before it gets to do that). In those cases, the task will stay in `.running` state until the network becomes available again, and only then will restart.  It achieves the desired effect, but is unfortunately not entirely consistent always.

**Comment by YankoSoulisimo** [comment]
@781flyingdutchman , hi, thanks for working on this, really appreciate it. I won't be able to test it in the next two weeks - I'm on vacation, but will do it as soon as I'm back. Thanks!

In the meantime - just wanted to share that similar weird behaviour is observed on Samsung devices with Android versions 12 and 13. Whenever my app goes in the background there, the download progress reports bigger, then smaller values - apparently, new tasks are enqueued and run in parallel again. I'm not able to share more details of this now because it doesn't happen in development environment, only in release builds tested by our QAs. I even tried to run with release config from Android Studio but with no success in reproducing.

**Comment by 781flyingdutchman** [comment]
Thanks and no worries - I'll publish it now in V7.5.0 as I am pretty confident it addresses the issue. I also think it will address the new issue with Samsung devices, as it sounds similar in nature (the Android OS reschedules the task, and so does the FileDownloader, leading to simultaneous downloads of the same file ad taskId, which leads to jumping progress), though I don't understand why the OS would cancel a task when the app moves to the background. If it does not fix the problem, please reopen this issue and let me know more details.

**Comment by 781flyingdutchman** [comment]
In V7.9.1 failed download tasks can now be resumed (under certain conditions) even if the Task.allowPause field is false. Resuming a failed task will attempt to continue the download where the failure occurred. If retries are set to a value >0 then retries will also first attempt to resume, and only start from scratch if that fails.

Tasks can only resume if the ETag header provided by the server is strong, and equal to the ETag at the moment the download was paused/failed, or if it is not provided at all.
